x/farming/spec/03_state_transitions.md

```diff
@@ -25,52 +25,57 @@ const (
-- When a farmer add/remove stakings to/from existing `Staking`, `StakedCoins` and `QueuedCoins` are updated in the corresponding `Staking`.
-- `QueuedCoins` : newly staked coins are in this status until end of current epoch, and then migrated to `StakedCoins` at the end of current epoch.
-- When a farmer unstakes, `QueuedCoins` are unstaked first, and then `StakedCoins`.
-
-## Reward Withdrawal
-
-To assume constant staking amount for reward withdrawal, automatic withdrawal is designed as below:
-- Add staking position : When there exists `QueuedCoins` in `Staking` at the end of current epoch
-  - accumulated rewards until current epoch are automatically withdrawn
-  - `StartEpochId` is modified to the `EpochId` of the next epoch
-  - `QueuedCoins` is migrated to `StakedCoins`
-- Remove staking position : When a farmer remove stakings from `StakedCoins`
-  - accumulated rewards until last epoch are immediately withdrawn
-  - `StartEpochId` is modified to the `EpochId` of the current epoch
-  - unstake executed immediately and `StakedCoins` are reduced accordingly
-- Manual reward withdrawal : When a farmer request a reward withdrawal
-  - accumulated rewards until last epoch are immediately withdrawn
-  - `StartEpochId` is modified to the `EpochId` of the current epoch
-
-## Accumulated Reward Calculation
-
-- Accumulated Unit Reward : AUR represents accumulated rewards(for each staking coin) of a staking position with amount 1.
-- AUR for each staking coin for each epoch can be calculated as below
-  - ![](https://latex.codecogs.com/svg.latex?\Large&space;\sum_{i=0}^{now}\frac{TR_i}{TS_i})
-    - ![](https://latex.codecogs.com/svg.latex?\Large&space;i) : each `EpochId`
-    - ![](https://latex.codecogs.com/svg.latex?\Large&space;now) : current `EpochId`
-    - ![](https://latex.codecogs.com/svg.latex?\Large&space;TS_i) : total staking amount of the staking coin for epoch i
-    - ![](https://latex.codecogs.com/svg.latex?\Large&space;TR_i) : total reward amount of the staking coin for epoch i
-- Accumulated rewards from any staking position can be calculated from AUR and the staking amount of the position as below
-  - ![](https://latex.codecogs.com/svg.latex?\Large&space;x*\(\sum_{i=0}^{now}\frac{TR_i}{TS_i}-\sum_{i=0}^{start}\frac{TR_i}{TS_i}\))
-    - assuming constant staking amount for the staking epochs
-    - ![](https://latex.codecogs.com/svg.latex?\Large&space;x) : staking amount for the staking period
\ No newline at end of file
+## Stake
+
+When a farmer stakes an amount of coins, the following state transitions occur:
+
+- it reserves the amount of coins to the staking reserve pool account `StakingReservePoolAcc` 
+- it creates `QueuedStaking` object and stores the staking coins in `QueueStaking`, which are waiting in a queue until the end of epoch to move to `Staking` object
+- it imposes more gas if the farmer already has `Staking` with the same coin denom
```

Comment:
It would be modified by https://github.com/tendermint/farming/pull/175/files#r730642438 @hallazzang